### DISCUSSION TITLE
Missing events when using Github Apps instead of personal token

### DISCUSSION THREAD

**Discussion by maxcleme** [community/community#59527] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

When using the following endpoint [/repos/{owner}/{repo}/issues/{issue_number}/events](https://docs.github.com/en/rest/issues/events?apiVersion=2022-11-28#list-issue-events) with my own token, I can see plenty of events, including some with [referenced](https://docs.github.com/en/webhooks-and-events/events/issue-event-types#referenced) type. eg:

```
{labeled 2023-06-22 18:07:12 +0000 UTC }
{review_requested 2023-06-22 18:07:12 +0000 UTC }
{review_requested 2023-06-22 18:07:12 +0000 UTC }
{review_requested 2023-06-22 18:07:12 +0000 UTC }
{labeled 2023-06-22 18:07:21 +0000 UTC }
{head_ref_force_pushed 2023-06-23 11:36:39 +0000 UTC }
{review_requested 2023-06-23 12:03:33 +0000 UTC }
{head_ref_force_pushed 2023-06-23 13:51:36 +0000 UTC }
{head_ref_force_pushed 2023-06-26 08:16:03 +0000 UTC }
{labeled 2023-06-26 10:30:02 +0000 UTC }
{labeled 2023-06-26 11:01:21 +0000 UTC }
{referenced 2023-06-26 11:01:27 +0000 UTC } <--
{referenced 2023-06-26 11:53:37 +0000 UTC } <--
{unlabeled 2023-06-26 11:55:26 +0000 UTC }
{unlabeled 2023-06-26 11:57:13 +0000 UTC }
{labeled 2023-06-26 12:03:46 +0000 UTC }
{referenced 2023-06-26 12:03:53 +0000 UTC } <--
{labeled 2023-06-26 12:03:58 +0000 UTC }
{referenced 2023-06-26 12:04:01 +0000 UTC } <--
{unlabeled 2023-06-26 12:21:10 +0000 UTC }
{merged 2023-06-26 13:22:30 +0000 UTC }
{closed 2023-06-26 13:22:31 +0000 UTC }
{unlabeled 2023-06-26 13:22:37 +0000 UTC }
{unlabeled 2023-06-26 13:22:38 +0000 UTC }
```

However, when using the same request but with Github Apps credentials, these events are mysteriously filtered (as well as some other for unknown reason):

```
{review_requested 2023-06-22 18:07:12 +0000 UTC }
{review_requested 2023-06-22 18:07:12 +0000 UTC }
{review_requested 2023-06-22 18:07:12 +0000 UTC }
{labeled 2023-06-22 18:07:21 +0000 UTC }
{head_ref_force_pushed 2023-06-23 11:36:39 +0000 UTC }
{review_requested 2023-06-23 12:03:33 +0000 UTC }
{head_ref_force_pushed 2023-06-23 13:51:36 +0000 UTC }
{head_ref_force_pushed 2023-06-26 08:16:03 +0000 UTC }
{labeled 2023-06-26 10:30:02 +0000 UTC }
{labeled 2023-06-26 11:01:21 +0000 UTC }
{unlabeled 2023-06-26 11:55:26 +0000 UTC }
{unlabeled 2023-06-26 11:57:13 +0000 UTC }
{labeled 2023-06-26 12:03:46 +0000 UTC }
{labeled 2023-06-26 12:03:58 +0000 UTC }
{unlabeled 2023-06-26 12:21:10 +0000 UTC }
{merged 2023-06-26 13:22:30 +0000 UTC }
{closed 2023-06-26 13:22:31 +0000 UTC }
```

I also tried with [/repos/{owner}/{repo}/issues/{issue_number}/timeline](https://docs.github.com/en/rest/issues/timeline?apiVersion=2022-11-28#list-timeline-events-for-an-issue) but the same filtering rules seems to be applied (works with personal token but not with Github Apps, events are missing)

Same goes with the following GraphQL query (works with personal token but not with Github Apps, events are missing):

```
query { 
  repository(owner: "aa", name:"bb") {
    pullRequest(number: 111) {
      timelineItems(itemTypes: REFERENCED_EVENT, last: 100) {
        edges {
          node
          {
            ... on ReferencedEvent {
              actor {
                login
              }
              commit {
                oid
              }
            }
          }
        }
      }
    }
  }
}
```

I can't find anything in the documentation stating this behavior, or at least explicitly calling our for required scopes in order to get specific types of events, hence the "bug" label.

**Comment by myhr-chman** [comment]
Thanks for starting this discussion, @maxcleme!

Since the last few weeks, I've also noticed something similar happening with `CROSS_REFERENCED_EVENT`s when using the GitHub CLI (`gh`) in a *GitHub workflow*. These events suddenly seemed to be filtered out and this seems to be limited *only to references from other private repositories*.

Couldn't pinpoint a problem till I saw this discussion.

```sh
# todo: need to replace `$ISSUE` with the node ID of any issue
gh api graphql --field issueId="$ISSUE" --raw-field query='
  query ($issueId: ID!) {
    node (id: $issueId) {
      ... on Issue {
        number,
        title,
        closed,
        closedAt,
        timelineItems (first: 20, itemTypes: [CROSS_REFERENCED_EVENT]) {
          nodes {
            ... on CrossReferencedEvent {
              source {
                ... on PullRequest {
                  number,
                  title,
                  closed,
                  closedAt,
                }
              },
              willCloseTarget,
            }
          }
        }
      }
    }
  }
'
```

Footnote: It seems like `gh` (which I've authenticated with OAuth) finds these events when I run the same command myself in a terminal.

Footnote: Unsure exactly if this is related to visibility or repository ownership.
